### ISSUE TITLE
Extra config options for `IDE0037`

### ISSUE THREAD

**Issue by lonix1** [dotnet/roslyn#60083]
**Brief description:**

The `IDE0037` analyser has config option [`dotnet_style_prefer_inferred_anonymous_type_member_names`](https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0037#dotnet_style_prefer_inferred_anonymous_type_member_names).

````cs
new {
  foo.Something,
  foo.Other,
  Value = foo.GetValue(),
  Name = bar.GetName(),
  baz.Id,
}
````

If the config option is `true` the above example passes. But it looks messy and is hard to read. It would be better to be explicit:

````cs
new {
  Something = foo.Something,
  Other     = foo.Other,
  Value     = foo.GetValue(),
  Name      = bar.GetName(),
  Id        = baz.Id,
}
````

But since the config option is `true`, the analyser will complain.

So it would be nice to have another option, `"trueForAll"` or something like that... I'm unsure of a good name. It works like `true`, unless there is at least one member which is explicitly named, in which case the analyser does nothing.

**Languages applicable:**
c#

**Code example that the analyzer should report:**
See above

**Additional information:**
No

**Documentation requirements:**

When this analyzer is implemented, it must be documented by following the steps at [Documentation for IDE CodeStyle analyzers](https://github.com/dotnet/roslyn/blob/main/docs/contributing/Documentation%20for%20IDE%20CodeStyle%20analyzers.md).

**Comment by sharwell** [comment]
💭 This seems like a good candidate for a third-party analyzer. Do we have evidence that the following is a common scenario?

1. The code has a significant number of anonymous types
2. The user wants anonymous types to have explicit property names
3. The user is unable to set `dotnet_style_prefer_inferred_anonymous_type_member_names = false`

**Comment by lonix1** [comment]
How would that work - you'd disable (`dotnet_style_prefer_inferred_anonymous_type_member_names = false`) this analyser and enable the other analyser?

**Comment by sharwell** [comment]
You'd set `dotnet_style_prefer_inferred_anonymous_type_member_names` to the option you use _most often_ (when features generate code they will use the value defined here) and set IDE0037 to either `hidden` or `none`. Then the other analyzer would be set to `warning` and handle reporting for all cases that do not meet the expected style.
